osquery/filesystem/tests/filesystem.cpp

```diff
@@ -892,9 +890,7 @@ TEST_F(FilesystemTests, test_directory_listing_with_recursive_junction) {
   std::vector<std::string> found_directories;
   ASSERT_TRUE(
       listDirectoriesInDirectory(test_root_raw, found_directories, false));
-  ASSERT_TRUE(!found_directories.empty());
-
-  ASSERT_TRUE(found_directories.size() == EXPECTED_NR_OF_DIRECTORIES);
+  ASSERT_TRUE(found_directories.empty());
```

Comment:
Why is this empty now?

Reply:
Because this is a link that points to itself. It cannot be resolved canonically, so it doesn't make sense to return it as a directory.